ats/atsMachines/fluxScheduled.py

```diff
@@ -0,0 +1,101 @@
+#ATS:slurm56 SELF FluxScheduled 3
+#ATS:toss_4_x86_64_ib SELF FluxScheduled 16
+
+from ats.atsMachines import lcMachines
+import time
+from math import ceil
+
+## Must be on a Flux machine or with Flux broker running in Slurm allocation for these imports to work.
+
+import flux
+import flux.job
+f = flux.Flux() ## defines a persistent handle to use to connect to the broker
+
+## Do we need to request an allocation? On fluke if you issue Flux mini run it sends to a compute node automatically
+## but there's no assurances it will be to the _same_ compute nodes for each of ats's "tests" or whatever
+## answer: YES for now
+
+## Should start its own Flux session!! TODO
+
+class FluxScheduled(lcMachines.LCMachineCore):
+    def init(self):
+        ## anything to initialize here?
+        ## set ceiling on number of nodes and cores in the allocation.
+        self.maxNodes = int(flux.resource.list.resource_list(f).get().free.nnodes)
+        self.maxCores = int(flux.resource.list.resource_list(f).get().free.ncores)
```

Comment:
Same as below, would be better to call resource_list once and save the result.

Reply:
Ah :)